src/Symfony/Component/Scheduler/Generator/MessageGenerator.php

```diff
@@ -72,8 +72,11 @@ public function getMessages(): \Generator
             }
 
             if ($yield) {
-                yield (new MessageContext($this->name, $id, $trigger, $time, $nextTime)) => $message;
-                $checkpoint->save($time, $index);
+                try {
+                    yield (new MessageContext($this->name, $id, $trigger, $time, $nextTime)) => $message;
+                } finally {
+                    $checkpoint->save($time, $index);
```

Comment:
Moving this line before the `yield` would do the same, right?

Reply:
No, that would the scheduler to create messages without interruption (tested)

Reply:
I didn't catch what wouldn't work, I added a test and it makes no difference.
Feel free to refine it to highlight the difference.

Reply:
> > No, that causes the scheduler to create messages without interruption (tested)
> 
> @maelanleborgne I didn't catch what wouldn't work, I added a test and it makes no difference. Feel free to refine it to highlight the difference.

Sorry I misread the comment. I tested on the [6.4 version of the component,](https://github.com/symfony/symfony/commit/06b16bd921cee96ff4c2738d647cf05d59aad0ad#diff-3431200d32b70ea151463d9c189f082bca9767fa7fbd128ed6d9646f05506df4R83) which uses à `foreach ($recurringMessage...)`, and there we need to wrap the `foreach` in a `try` .
That will be good to know when merging up 6.3 into 6.4

Reply:
yes, I thought about it, because the related issue was on 6.4. 
So for 6.3, shouldn't we keep the `try/finally` ?

Reply:
Not sure what's the recommended way of doing it.
As suggested by @fabpot, the try/finally isn't necessary in 6.3, so we might as well no overcomplicate the code and just move the `$chekpoint->save()` above the `yield`.
Then you could just keep an eye on the 6.4 and 7 branches to see how it reflects once merged up. And open a new PR if needed :smiley:

Reply:
Yes, that makes sense.
I couldn't make a choice on my own, both solutions seem equivalent, so maybe the core team could help us ? 😃

Reply:
Let's keep the `try/finally` block as it will be more consistent with the code in 6.4/7.0.